tested/languages/java/templates/value_basic.mako

```diff
@@ -4,7 +4,12 @@
 <%! from json import dumps %>\
 <%page args="value" />\
 % if value.type == BasicNumericTypes.INTEGER:
-    ${value.data}\
+    ## Basic heuristic for long numbers
+    % if len(str(value.data)) > 9:
```

Comment:
Can you just check for the actual number?
```suggestion
    % if value.data > (2**31 - 1):
```

Reply:
Good point actually!